### PR TITLE
Fixed an issue where categories were not updated in the App Store aft…

### PR DESCRIPTION
**Pull request by king-119**
…er the app template was republished

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
Fixed an issue that category of application can not be updated.
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/models/openpitrix/applicationversions.go [hunk 1]
```diff
@@ -428,7 +428,8 @@ func (c *applicationOperator) DoAppVersionAction(versionId string, request *Acti
 			return err
 		}
 
-		if !reflect.DeepEqual(&app.Spec, &appInStore.Spec) {
+		if !reflect.DeepEqual(&app.Spec, &appInStore.Spec) || !reflect.DeepEqual(app.Labels[constants.CategoryIdLabelKey], appInStore.Labels[constants.CategoryIdLabelKey]) {
+			appCopy.Labels[constants.CategoryIdLabelKey] = app.Labels[constants.CategoryIdLabelKey]
 			appCopy := appInStore.DeepCopy()
```
Comment (hongzhouzi) [inline pkg/models/openpitrix/applicationversions.go:433]:
```
appCopy := appInStore.DeepCopy()
appCopy.Labels[constants.CategoryIdLabelKey] = app.Labels[constants.CategoryIdLabelKey]
```

**Review by hongzhouzi — CHANGES_REQUESTED** [review]

File: pkg/models/openpitrix/applicationversions.go (hunk 1 above)
Comment (hongzhouzi) [reply pkg/models/openpitrix/applicationversions.go:433]: The appCopy variable is declared in the following line, so the verification fails. You need to adjust the code position.

Comment (king-119) [reply pkg/models/openpitrix/applicationversions.go]: Sorry, the position of the code was added to the wrong position, we have made corrections and resubmitted the code

**Pull request merged**